### PR TITLE
Fix goal, steps and plan types

### PR DESCRIPTION
**Pull request by Thomas-Geraghty**
- We really need to sort out types out
- Fixed a bunch of types to match the API
- Fixed and updated test data (I hate this test data thing now!)

**Note: I think NewStep and Step could be made into one file, same with NewGoal and Goal - but it would end up changing a bunch of other file imports so worth doing as a separate PR**

### CODE REVIEW

File: server/@types/GoalType.ts [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
+  areaOfNeed: AreaOfNeed
+  relatedAreasOfNeed: AreaOfNeed[]
   creationDate: string
   completedDate?: string
   targetDate: string
-  isAgreed?: boolean
-  agreementNote: string
   goalOrder?: number
   moveUpURL?: string
   moveDownURL?: string
```
Comment (Thomas-Geraghty) [inline server/@types/GoalType.ts:11]: I don't get why these are part of the `Goal` object? This data is only used for one view, it shouldn't have leaked into the common object.

File: server/@types/GoalType.ts [hunk 2]
```diff
@@ -1,14 +1,17 @@
...
 }
+
+type AreaOfNeed = {
+  uuid: string
```
Comment (Thomas-Geraghty) [inline server/@types/GoalType.ts:15]: Also we shouldn't be returning the UUID from the area of need to the view/frontend.

File: server/@types/NewStepType.ts [hunk 3]
```diff
@@ -1,5 +1,10 @@
...
   description?: string
-  actor: string[]
+  actor: Actor[]
   status?: string
 }
+
+type Actor = {
+  actor: string
+  actorOptionId: number
+}
```
Comment (Thomas-Geraghty) [inline server/@types/NewStepType.ts:10]: If we combined NewStepType and StepType, this type could be shared across both 😄 - one for the future

**Review by pk2453 — APPROVED** [review]

File: server/@types/GoalType.ts (hunk 1 above)
Comment (pipwilson) [reply server/@types/GoalType.ts:11]: Moved into https://github.com/ministryofjustice/hmpps-sentence-plan-ui/pull/76

**Pull request merged**